packages/@internationalized/date/tests/conversion.test.js

```diff
@@ -138,6 +139,35 @@ describe('CalendarDate conversion', function () {
     });
   });
 
+  describe('setTimeZoneOffset', () => {
+    it('should support old dates in local timezone with second offsets', () => {
+      let resolvedOptions = Intl.DateTimeFormat.prototype.resolvedOptions;
+      const testCases = [
+        ['America/New_York', '1800-01-01T00:00:00Z', -17_762_000],
+        ['Europe/London', '1800-01-01T00:00:00Z', -75_000],
+        ['Europe/Berlin', '1800-01-01T00:00:00Z', 3_208_000],
+        ['Europe/Rome', '1800-01-01T00:00:00Z', 2_996_000]
+      ];
+      for (let [timezone, date, expectedOffset] of testCases) {
+        // Unfortunate: TZ env var determines Date.getTimezoneOffset()
+        // but overrides local timezone for parallell tests.
+        process.env.TZ = timezone;
+        jest.spyOn(Intl.DateTimeFormat.prototype, 'resolvedOptions').mockImplementation(function () {
+          let s = resolvedOptions.call(this);
+          s.timeZone = timezone;
+          return s;
+        });
+
+        const tzOffset = getTimeZoneOffset(new Date(date).getTime(), timezone);
+        expect(tzOffset).toBe(expectedOffset);
+
+        jest.clearAllMocks();
+        clearLocalTimeZoneCache();
+      }
+      delete process.env.TZ;
```

Comment:
Can't we just do something like this? or what am I overlooking? it succeeds with your changes and fails without your change.

```suggestion
      for (let [timezone, date, expectedOffset] of testCases) {
        jest.spyOn(Intl.DateTimeFormat.prototype, 'resolvedOptions').mockImplementation(function () {
          let s = resolvedOptions.call(this);
          s.timeZone = timezone;
          return s;
        });

        const tzOffset = getTimeZoneOffset(new Date(date).getTime(), timezone);
        expect(tzOffset).toBe(expectedOffset);

        jest.clearAllMocks();
      }
```

Reply:
I updated the test according to your suggestions :+1:

I come short at communicating the problems with this. But I agree that testing this edge case has some value as documentation.

- If you want to discover the issues, you can add test cases for 2 different timezones (with date > 1970).
- If you want to discuss how this can be tested, you can send me an email to set up a video chat 😄